### PR TITLE
#42 changes for reading crysl rules in source code format.

### PR DESCRIPTION
**Pull request by gokcekrky**
#42 changes for reading crysl rules in source code format.
Crysl rules in resources folder should be in crysl source code format.

### CODE REVIEW

File: CryptoAnalysis/src/main/java/crypto/analysis/CrySLRulesetSelector.java [hunk 1]
```diff
@@ -15,7 +16,11 @@
 	public static List<CryptSLRule> makeFromRuleset(String rulesBasePath, Ruleset... set) {
 		List<CryptSLRule> rules = Lists.newArrayList();
 		for (Ruleset s : set) {
-			rules.addAll(getRulesset(rulesBasePath, s));
+			//rules.addAll(getRulesset(rulesBasePath, s));
```
Comment (kruegers) [inline CryptoAnalysis/src/main/java/crypto/analysis/CrySLRulesetSelector.java]: Please clean up the code.

File: CryptoAnalysis/src/main/java/crypto/HeadlessCryptoScanner.java [hunk 2]
```diff
@@ -72,7 +72,11 @@ public static HeadlessCryptoScanner createFromOptions(String... args) throws Par
 
 		if (options.hasOption("rulesDir")) {
 			String resourcesPath = options.getOptionValue("rulesDir");
-			rules = CrySLRulesetSelector.makeFromPath(new File(resourcesPath));
+			//rules = CrySLRulesetSelector.makeFromPath(new File(resourcesPath));
```
Comment (kruegers) [inline CryptoAnalysis/src/main/java/crypto/HeadlessCryptoScanner.java]: Please clean up the code.

File: CryptoAnalysis/src/test/java/test/IDEALCrossingTestingFramework.java [hunk 3]
```diff
@@ -71,8 +71,9 @@ public CrySLResultsReporter analysisListener() {
 	}
 
 	protected CryptSLRule getRule() {
-		return CrySLRulesetSelector.makeSingleRule(RULES_BASE_DIR, getRuleset(), getRulename());
-//		CryptSLRuleReader.readFromFile(file);
+		//return CrySLRulesetSelector.makeSingleRule(RULES_BASE_DIR, getRuleset(), getRulename());
```
Comment (kruegers) [inline CryptoAnalysis/src/test/java/test/IDEALCrossingTestingFramework.java]: Please clean up the code.

**Comment by johspaeth** [comment]
Thanks for working on that. 

Do we need to distribute the `.jar` in the repository? Can't we use the default maven dependency load mechanism by declaring the artifact and version in the `pom.xml` such that it downloads the latest version from a remote repository.

The downside of your approach is that we have to manually update the `jar? file upon changes, which I would like to avoid, if possible.

**Comment by kruegers** [comment]
Hey @johspaeth,
I was waiting for you to give your o.k. to merge this pull request as well. Are you perhaps waiting for me to just merge it? Just wanting to make sure, so we wouldn't keep waiting for each other. :)

**Comment by gokcekrky** [comment]
Last changes in cryptsl (pull request # 19) is required for the solution of jvmtype error.

**Comment by johspaeth** [comment]
@gokcekrky I have tested the branch locally on my machine and it works, therefore I merge. Thanks for adding this feature.  (The CI fails due to reasons regarding an outdated rule set).

Unfortunately, the build time now changed from 6min to 26min which we do not want to maintain in the long run. Could you please investigate options how to reduce test execution time?

I assume initializing the rules sets only once globally could be enough. 

Additionally, please explore options to get rid of the logging

> 123771 [main] INFO  t.linking.impl.ImportedNamesAdapter  - getElements should be called with a QualifiedName during linking.

**Comment by johspaeth** [comment]
@gokcekrky I have tested the source code reading within the Eclipse plugin. The we could eventually get rid of using .cryptslbin files at all. Unfortunately, the setup time is also way too long in the Eclipse environment and we cannot yet use it as default within the plugin: It takes about 30 seconds to show a warning in a simple test program. 

Do you have an idea what decreases the analysis time that much? Please do some profiling.

**Comment by gokcekrky** [comment]
@johspaeth Unfortunately I couldn't test it within the Eclipse plugin because of compiling error. 
I am planning to initialize the rules sets only once as you suggested. It may help to decrease analysis time in Eclipse environment.

**Comment by gokcekrky** [comment]
@johspaeth I've implemented regarding changes and created pull request #170 .
The changes for undesirable logs are implemented on cryptsl side, please see pull request 21 on the cryptsl repository.

**Pull request merged**